apollo-federation/src/sources/connect/expand/mod.rs

```diff
@@ -393,32 +394,113 @@ mod helpers {
                 )]),
             };
 
+            // The input arguments are independent of the actual output type in which they are used, but
+            // require access to the field of the output in order to process the inputs. The output type
+            // needs to be present in order to modify its keys and other directives, so we split up
+            // the processing so that each has what it needs before and after adding the field.
+            self.process_inputs(to_schema, connector, &field_def.arguments)?;
             parent.insert(to_schema, Node::new(connector_parent_type))?;
-
-            // Now we need to process any inputs that need to be carried over, making sure to keep track
-            // of how these inputs should influence the output types in terms of federation directives.
-            self.process_inputs(
-                to_schema,
-                connector,
-                &field_def.arguments,
-                self.key_name.clone(),
-                parent_type.name.clone(),
-                output_name,
-            )?;
+            self.process_outputs(to_schema, connector, parent_type.name.clone(), output_name)?;
```

Comment:
i kinda want a different name for process_outputs — we're processing the _output_ types related to the _inputs_ (keys) of the entity resolver. but maybe it's just a matter of documentation!